### DISCUSSION TITLE
zh/models/yolov9/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8888] [Docs]
# zh/models/yolov9/

了解 YOLOv9，它是实时物体检测系统的最新成员，利用可编程梯度信息和 GELAN 架构实现了无与伦比的性能。

https://docs.ultralytics.com/zh/models/yolov9/

**Comment by XushaopengHNU** [comment]
Please upload the download link of yolov9c.yaml, because the yolov9-c.yaml downloaded on the github project of the yolov9 author gives an error after running, and it seems to do not match the code in this page.

**Reply by glenn-jocher** [reply]
Hi there! 👋 It seems like you're encountering issues with the `yolov9-c.yaml` file. To ensure compatibility and resolve the error you're experiencing, I recommend checking the following:

1. Ensure you're using the latest version of the YOLOv9 codebase from the official GitHub repository. Sometimes, discrepancies arise due to updates or changes in the code structure.
2. Verify that the `yolov9-c.yaml` file is correctly placed in the expected directory as per the YOLOv9 documentation. The file path should match the one expected by the script or command you're running.
3. If the issue persists, consider manually adjusting the `yolov9-c.yaml` to match the configuration details mentioned on the [YOLOv9 documentation page](https://docs.ultralytics.com/zh/models/yolov9/). Sometimes, minor edits in the YAML file can resolve compatibility issues.

If you're still facing difficulties, could you please share the specific error message or describe the issue in more detail? This will help in providing a more targeted solution. 😊

Here's a quick example on how to load a YOLOv9 model in Python, assuming the YAML and model files are correctly configured:

```python
from ultralytics import YOLO

# Load a YOLOv9 model from a YAML configuration file
model = YOLO('path/to/yolov9c.yaml')

# Optionally, load a pretrained model
model = YOLO('path/to/yolov9c.pt')
```

Feel free to reach out if you need further assistance!

**Comment by ZSTHL** [comment]
Can anyone help me?
When i use my training best.pt in ultralytics, it has an error


Traceback (most recent call last):
  File "E:\yolov9-main\test.py", line 4, in <module>
    results=model(img)
  File "E:\python3.9.13\lib\site-packages\ultralytics\engine\model.py", line 176, in __call__
    return self.predict(source, stream, **kwargs)
  File "E:\python3.9.13\lib\site-packages\ultralytics\engine\model.py", line 445, in predict
    self.predictor.setup_model(model=self.model, verbose=is_cli)
  File "E:\python3.9.13\lib\site-packages\ultralytics\engine\predictor.py", line 297, in setup_model
    self.model = AutoBackend(
  File "E:\python3.9.13\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "E:\python3.9.13\lib\site-packages\ultralytics\nn\autobackend.py", line 144, in __init__
    model = model.fuse(verbose=verbose)
TypeError: fuse() got an unexpected keyword argument 'verbose

**Reply by glenn-jocher** [reply]
It looks like you encountered an issue with the `fuse()` method not accepting a `verbose` argument when trying to use your trained model. This issue could be due to a version mismatch between the Ultralytics library and your model or script. Here are a couple of steps you could try to resolve the issue:

1. **Update Ultralytics Library**: Ensure that you are using the latest version of the Ultralytics library. You can update it using pip:
   ```bash
   pip install --upgrade ultralytics
   ```

2. **Check Compatibility**: If your model was trained with a specific version of the Ultralytics library, make sure that your current environment matches this version to avoid compatibility issues.

If these steps don't solve the problem, could you please share more details about how the model was trained (e.g., library version, custom modifications)? That could help in diagnosing the issue further. 😊

**Comment by YJ-2017** [comment]
您好，本页面下载的YOLOv9e模型文件，对应的是yolov9仓库中的gelan-e还是yolo9-e-converted，能否提供完整的yolo9-e (GELAN+PGI)模型文件？

**Reply by pderrenger** [reply]
@YJ-2017 您好，

感谢您对YOLOv9的关注和提问！关于您提到的模型文件，YOLOv9e模型对应的是完整的YOLOv9-e (GELAN+PGI)模型文件。您可以在我们的[YOLOv9文档页面](https://docs.ultralytics.com/models/yolov9/)找到更多详细信息和下载链接。

如果您在使用过程中遇到任何问题，建议您提供一个可复现的示例代码，这样我们可以更有效地帮助您解决问题。您可以参考这个[最小可复现示例指南](https://docs.ultralytics.com/help/minimum_reproducible_example)来创建您的示例。

另外，请确保您使用的是最新版本的相关包，以避免已解决的问题再次出现。

希望这些信息对您有所帮助！如果还有其他问题，请随时告知。

祝好！ 😊

Ultralytics 团队

**Reply by YJ-2017** [reply]
yolov9仓库中gelan-e文件大小为114547KB，yolov9-e-converted文件大小为114415KB，YOLOv9-e 文件大小为136932KB，yolov9-e-converted是由YOLOv9-e删除了辅助分支（PGI）之后的模型权重文件。
ultralytics提供的YOLOv9e模型文件大小为114769KB（与gelan-e、yolov9-e-converted文件大小差不多），因此有如上疑惑，还望解答。

**Reply by pderrenger** [reply]
您好，

感谢您对YOLOv9的关注和详细问题描述！关于您提到的模型文件大小差异，确实可能会引起一些疑惑。以下是一些可能的解释：

1. **模型文件大小差异**：不同版本的模型文件大小可能会有所不同，具体取决于模型中包含的层数、参数量以及是否包含辅助分支（如PGI）。您提到的YOLOv9-e模型文件大小为136932KB，而我们提供的YOLOv9e模型文件大小为114769KB，这表明我们提供的模型可能是经过优化或简化的版本。

2. **辅助分支（PGI）**：如果模型文件删除了辅助分支（PGI），其大小会有所减少。我们提供的模型文件可能已经进行了这种优化，以提高模型的推理速度和效率。

为了更好地帮助您解决问题，建议您提供一个最小可复现示例代码，这样我们可以更有效地诊断和解决问题。您可以参考这个[最小可复现示例指南](https://docs.ultralytics.com/help/minimum_reproducible_example)来创建您的示例。

另外，请确保您使用的是最新版本的相关包，以避免已解决的问题再次出现。如果问题仍然存在，请随时告知，我们会尽力帮助您解决。

希望这些信息对您有所帮助！如果还有其他问题，请随时告知。

祝好！ 😊

Ultralytics 团队
